casbin/enforcer.cpp

```diff
@@ -444,6 +444,40 @@ bool Enforcer :: Enforce(Scope scope) {
     return this->enforce("", scope);
 }
 
+// Enforce with two params, decides whether a "subject" can do the operation "action", input parameters are usually: (sub, act).
+bool Enforcer::Enforce(string sub, string act) {
+    return Enforce({ sub,act });
+}
+
+// Enforce with three params, decides whether a "subject" can access a "object" with the operation "action", input parameters are usually: (sub, obj, act).
+bool Enforcer::Enforce(string sub, string obj, string act) {
+    return Enforce({sub,obj,act});
+}
+
+// Enforce with four params, decides whether a "subject" can access a "object" with the operation "action" in the domain "dom", input parameters are usually: (sub, dom, obj,act).
+bool Enforcer::Enforce(string sub, string dom, string obj, string act) {
+    return Enforce({ sub,dom,obj,act });
+}
+
+// Enforce with a vector param,decides whether a "subject" can access a "object" with the operation "action", input parameters are usually: (sub, obj, act).
+bool Enforcer::Enforce(vector<string> params) {
+    vector <string> r_tokens = this->model->m["r"].assertion_map["r"]->tokens;
+
+    int r_cnt = r_tokens.size();
+    int cnt = params.size();
+
+    if (cnt != r_cnt)
+        return false;
```

Comment:
return false or throw exception?

Reply:
> return false or throw exception?

I think it should throw exception.But to keep style with other parts of casbin-cpp, I return false instead of it.
We can add throwing exception to casbin-cpp in another PR.

Reply:
How is Golang handling this?

Reply:
Golang handles this by returning two values, false and the error, but as casbin-cpp follows the convention of returning only false, and logging the error message. But the logger is not developed yet, so we are not going to log the message for now.